### PR TITLE
Added an option to pack acceleration structures in one buffer

### PR DESCRIPTION
**Pull request by naomipappe**
- Added an option to pack the created acceleration structures in one buffer
- Moved the querying of the acceleration structure function pointers to common code

### CODE REVIEW

**Comment by per-mathisen-arm** [comment]
Please squash the commits into one and fix the added whitespace. Indented variable blocks are a nightmare to maintain.

File: src/vulkan_as_2.cpp [hunk 1]
```diff
@@ -115,38 +116,17 @@ Buffer prepare_buffer(const vulkan_setup_t& vulkan, VkDeviceSize size, void *dat
 	return buffer;
 }
 
-VkDeviceAddress get_buffer_device_adress(VkDevice device, VkBuffer buffer)
+VkDeviceAddress get_buffer_device_adress(const vulkan_setup_t& vulkan, VkBuffer buffer)
```
Comment (per-mathisen-arm) [inline src/vulkan_as_2.cpp]: I think this function too could be put into common code.

Comment (naomipappe) [reply src/vulkan_as_2.cpp]: I agree. I squashed the commits, and put the requested functionality into the common code, under `acceleration_structures` namespace.

File: src/vulkan_common.h [hunk 2]
```diff
@@ -50,6 +50,7 @@ struct vulkan_setup_t
 	PFN_vkSetDebugUtilsObjectNameEXT vkSetDebugUtilsObjectName = nullptr;
 	PFN_vkCmdInsertDebugUtilsLabelEXT vkCmdInsertDebugUtilsLabel = nullptr;
 	PFN_vkFrameEndTRACETOOLTEST vkFrameEnd = nullptr;
+	PFN_vkGetBufferDeviceAddress vkGetBufferDeviceAddress;
```
Comment (per-mathisen-arm) [inline src/vulkan_common.h]: Please set default value to nullptr.

**Review by per-mathisen-arm — APPROVED** [review]

File: src/vulkan_common.h (hunk 2 above)
Comment (naomipappe) [reply src/vulkan_common.h]: Implemented the requested changes.

**Pull request merged**